### DISCUSSION TITLE
UBL with BLTouch - inconsistent first layer

### DISCUSSION THREAD

**Discussion by hayhay7789** [Jyers/Marlin#946] [Q&A] [answered]
Hey everyone, so I'm using a custom firmware for my Ender 3 v2 with BLTouch and Micro-Swiss DD. I've been having a weird problem where my bed levelling doesn't seem to be following the topology of my bed. I have my start code set up to run a levelling routine at the start of each print, which gets to around probe 35/64 before starting. The first layer is always very inconsistent and has caused a LOT of my prints to fail recently (3/5 at least), with the center-right of the print appearing to have a Z-offset too low while the top left has a Z-offset too low. In other words, some parts of my print end up getting squashed into the bed and others barely adhere due to the Z distance. 

I have tried increasing the number of probe points and increasing the bed area accessible by the probe (hence the max X pos of 265 and probing margin of 2), as well as TONS of manual levelling, but thus far I have had no success. The bed topology displayed in the graph on the LCD appears to be very level with very few inconsistencies, and no larger deviation than 0.12. Do I need to add auto-tilting the mesh to my start code or something?

Thanks, hopefully someone smart has an answer :) My config files and startcode are below:

```
; Ender 3 Custom Start G-code
G92 E0 ; Reset Extruder
M190 S{material_bed_temperature} ; check bed temp before running
G28 ; homes all axes 
G29 P1 ; run autoprobing UBL
G29 P3 ; extrapolates levelling to unreachable areas
G29 F10 ; fade height 10mm
G29 S1 ; save bed mesh to slot 1
G29 A ; activate UBL
G29 L1 ; load mesh 1
M500 ; save settings to EEPROM
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X2.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X2.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X2.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X2.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish
```

[Configs.zip](https://github.com/Jyers/Marlin/files/6573738/Configs.zip)

**Comment by kuroyume-cl** [comment]
I'm having a similar issue, where I can see that there is some adjustment is made but the nozzle is still too close in some places and too far in others.

**Reply by hayhay7789** [reply]
Sounds like we're in the same boat

**Comment by Jyers** [comment, marked as answer]
Have you tried not leveling right before your print? Perhaps try from the leveling menu and use the mesh viewer to check that your mesh seems accurate. Generally with UBL you don't probe before every print, you simply auto tilt.

**Reply by kuroyume-cl** [reply]
I load the mesh and tilt and still have some serious variability between four corners and the center. 
This is what I'm doing before every print:

`G28 ;Home`
`G29 A ;Start UBL`
`G29 L0 ;Load Mesh`
`G29 J2 ;Tilt Mesh using 4 points`

**Reply by Jyers** [reply]
You could try putting your G29 A after your tilt command. If you had the mesh disabled when you saved it it will also be disabled when you load it. Besides that I can't really imagine what could be causing the issue besides a physical problem

**Reply by kuroyume-cl** [reply]
Thanks, I've been trying to chase it as a physical issue so far so it's good to see it's gotta be that.

**Reply by hayhay7789** [reply]
Will give that a try ASAP. Thank you!
